### PR TITLE
DRAFT: add Maven

### PR DESCRIPTION
**Pull request by AMatutat**
@all pls verify

### CODE REVIEW

**Review by Kushurando — APPROVED** [review]

**Comment by AMatutat** [comment]
after second review please assign to me

**Comment by cagix** [comment]
@AMatutat No need to rebase this branch. I tried it, but there are servere merge conflicts :-/  However, it affects just the maven config, so no need to rebase in the end :-)

**Comment by AMatutat** [comment]
We merged #65 so i guess this need some adjustments. 

*Note: Next time we want to change the project structure, we should do it BEFORE we write build scripts*

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,14 @@
+## maven
```
Comment (cagix) [inline .gitignore:1]: now we have two maven sections? this one and another one below? please merge!

File: README.md [hunk 2]
```diff
@@ -10,6 +10,14 @@ lang: de-DE
 
 Dieses Dokument liefert einen Einstieg in das PM-Dungeon. Es erläutert die Installation der API und erläutert Ihnen die ersten Schritte, um eigene Inhalte zum Dungeon hinzuzufügen. Es dient als Grundlage für alle weiteren Praktika. Lesen Sie das Dokument daher aufmerksam durch und versuchen Sie sich zusätzlich selbst mit dem Aufbau vertraut zu machen.
 
+## Build jar mit Maven
+
+Die erforderliche jar kann einfach über Maven erstellt werden. Hierfür kann der folgende Befehl genutzt werden:
```
Comment (cagix) [inline README.md:15]: s/jar/`jar`/

File: README.md [hunk 3]
```diff
@@ -10,6 +10,14 @@ lang: de-DE
...
+```bash
+    mvn package
```
Comment (cagix) [inline README.md:17]: remove indentation

File: README.md [hunk 4]
```diff
@@ -10,6 +10,14 @@ lang: de-DE
...
+```
+Die jar mit allen Abhängigkeiten inkludiert liegt nun unter *"desktop/target/pmdungeon-desktop-jar-with-dependencies.jar"* und kann nach belieben umbenannt werden.
```
Comment (cagix) [inline README.md:19]: jar => `jar`

*"desktop/target/pmdungeon-desktop-jar-with-dependencies.jar"* => "`desktop/target/pmdungeon-desktop-jar-with-dependencies.jar`"

File: core/pom.xml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+		<dependency>
+			<groupId>com.google.code.gson</groupId>
+			<artifactId>gson</artifactId>
+			<version>${gson.version}</version>
+			<scope>compile</scope>
+		</dependency>
+	</dependencies>
+
+	<build>
+		<finalName>pmdungeon-core</finalName>
```
Comment (cagix) [inline core/pom.xml:50]: do we really have still `core`? iirc this got merged ...

File: core/pom.xml [hunk 6]
```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+	xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+	<modelVersion>4.0.0</modelVersion>
+	<artifactId>core</artifactId>
```
Comment (cagix) [inline core/pom.xml:7]: do we really have still `core`? iirc this got merged ...

File: desktop/pom.xml [hunk 7]
```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+	xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+	<modelVersion>4.0.0</modelVersion>
+	<artifactId>desktop</artifactId>
```
Comment (cagix) [inline desktop/pom.xml:7]: do we really have still `desktop`? iirc this got merged ...

File: desktop/pom.xml [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+		<dependency>
+			<groupId>com.badlogicgames.gdx</groupId>
+			<artifactId>gdx-freetype-platform</artifactId>
+			<version>${gdx.version}</version>
+			<classifier>natives-desktop</classifier>
+			<scope>compile</scope>
+		</dependency>
+		<dependency>
+			<groupId>de.pm-dungeon</groupId>
+			<artifactId>core</artifactId>
```
Comment (cagix) [inline desktop/pom.xml:46]: do we really have still `core`? iirc this got merged ...

File: desktop/pom.xml [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+			<version>${pmd-core.version}</version>
+			<scope>compile</scope>
+		</dependency>
+	</dependencies>
+
+	<build>
+		<finalName>pmdungeon-desktop</finalName>
```
Comment (cagix) [inline desktop/pom.xml:53]: do we really have still `desktop`? iirc this got merged ...

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+	<groupId>de.pm-dungeon</groupId>
+	<artifactId>pmdungeon</artifactId>
+	<version>1.0</version>
+	<packaging>pom</packaging>
+	<modules>
+		<module>core</module>
+		<module>desktop</module>
+	</modules>
+
+
```
Comment (cagix) [inline pom.xml:16]: one blank line should suffice

File: pom.xml [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+				<artifactId>gdx</artifactId>
+				<version>${gdx.version}</version>
+				<scope>compile</scope>
+			</dependency>
+		</dependencies>
+	</dependencyManagement>
+
+
+
+
```
Comment (cagix) [inline pom.xml:30]: one blank line should suffice

File: pom.xml [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+							<goals>
+								<goal>single</goal>
+							</goals>
+						</execution>
+					</executions>
+				</plugin>
+			</plugins>
+		</pluginManagement>
+	</build>
+</project>
```
Comment (cagix) [inline pom.xml:92]: no line ending here?

**Review by cagix — CHANGES_REQUESTED** [review]
well, this needs to be rebased and reworked: we don't have `core` and `desktop` anymore ...

**Comment by Kushurando** [comment]
I apparently can't revoke my approval from before, so don't count that for now, obviously

**Review by Tobias321 — CHANGES_REQUESTED** [review]
> $ git checkout upstream/Maven
> Note: switching to 'upstream/Maven'.
>
> You are in 'detached HEAD' state. 🙁 

There are some changes that needs to be in here. We don't longer have the split into core and desktop. Pls, update this PR and note cagix's review.

**Comment by AMatutat** [comment]
#76

**Pull request closed without merging**